/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 1167, in run
    pubsub.get_message(ignore_subscribe_messages=True, timeout=sleep_time)
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 1029, in get_message
    response = self.parse_response(block=(timeout is None), timeout=timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 840, in parse_response
    response = self._execute(conn, try_read)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 816, in _execute
    return conn.retry.call_with_retry(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/retry.py", line 65, in call_with_retry
    fail(error)
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 818, in <lambda>
    lambda error: self._disconnect_raise_connect(conn, error),
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 805, in _disconnect_raise_connect
    raise error
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/retry.py", line 62, in call_with_retry
    return do()
           ^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 817, in <lambda>
    lambda: command(*args, **kwargs),
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 834, in try_read
    if not conn.can_read(timeout=timeout):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/connection.py", line 490, in can_read
    return self._parser.can_read(timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/_parsers/hiredis.py", line 81, in can_read
    return self.read_from_socket(timeout=timeout, raise_on_timeout=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/_parsers/hiredis.py", line 92, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 1167, in run
    pubsub.get_message(ignore_subscribe_messages=True, timeout=sleep_time)
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 1029, in get_message
    response = self.parse_response(block=(timeout is None), timeout=timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 840, in parse_response
    response = self._execute(conn, try_read)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 816, in _execute
    return conn.retry.call_with_retry(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/retry.py", line 65, in call_with_retry
    fail(error)
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 818, in <lambda>
    lambda error: self._disconnect_raise_connect(conn, error),
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 805, in _disconnect_raise_connect
    raise error
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/retry.py", line 62, in call_with_retry
    return do()
           ^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 817, in <lambda>
    lambda: command(*args, **kwargs),
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 834, in try_read
    if not conn.can_read(timeout=timeout):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/connection.py", line 490, in can_read
    return self._parser.can_read(timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/_parsers/hiredis.py", line 81, in can_read
    return self.read_from_socket(timeout=timeout, raise_on_timeout=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/_parsers/hiredis.py", line 92, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:227: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return calendar.timegm(datetime.datetime.utcnow().utctimetuple())
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/rq/utils.py:128: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
  return datetime.datetime.utcnow()
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 1167, in run
    pubsub.get_message(ignore_subscribe_messages=True, timeout=sleep_time)
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 1029, in get_message
    response = self.parse_response(block=(timeout is None), timeout=timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 840, in parse_response
    response = self._execute(conn, try_read)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 816, in _execute
    return conn.retry.call_with_retry(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/retry.py", line 65, in call_with_retry
    fail(error)
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 818, in <lambda>
    lambda error: self._disconnect_raise_connect(conn, error),
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 805, in _disconnect_raise_connect
    raise error
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/retry.py", line 62, in call_with_retry
    return do()
           ^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 817, in <lambda>
    lambda: command(*args, **kwargs),
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/client.py", line 834, in try_read
    if not conn.can_read(timeout=timeout):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/connection.py", line 490, in can_read
    return self._parser.can_read(timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/_parsers/hiredis.py", line 81, in can_read
    return self.read_from_socket(timeout=timeout, raise_on_timeout=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vaishnavi/Documents/frappe/user_authentication_bench/env/lib/python3.12/site-packages/redis/_parsers/hiredis.py", line 92, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
